hooks/useMovieSearch.ts

```diff
@@ -39,21 +39,21 @@ const fetchMovies = async ({
   };
 };
 
-export const useMovieSearch = (query: string) => {
+export const useMovieSearch = (query: string, currentPage: number) => {
   const trimmedQuery = query.trim();
 
   return useInfiniteQuery({
     queryKey: [Queries.SEARCH_MOVIES, trimmedQuery],
     queryFn: fetchMovies,
     enabled: !!trimmedQuery,
-    initialPageParam: 1,
+    initialPageParam: currentPage,
     getNextPageParam: (lastPage) => {
       return lastPage.nextPage <= lastPage.totalPages
         ? lastPage.nextPage
         : undefined;
     },
     refetchOnWindowFocus: false,
     refetchOnReconnect: false,
-    staleTime: 1000 * 60 * 5,
+    staleTime: Infinity,
```

Comment:
I understand the need to use Infinity, however this will be a problem for online use, the information will be state as react-query won't refetch

Reply:
@pedrozok you're right. I actually set it this way since there's no cacheTime prop: [https://tanstack.com/query/latest/docs/framework/react/reference/useQuery](https://tanstack.com/query/latest/docs/framework/react/reference/useQuery). (useInfiniteQuery options are the same as useQuery). But you're right that this way there would be no refetching - data is always fresh.
I'm now reading more on this and the most similar prop to cacheTime would be gcTime, which when set to Infinity will prevent garbage collection (although this is maxed at 24 days). This means that by setting:

```
    staleTime: 1000 * 60 * 5, // Does regular (5mins) refetching, for online access
    gcTime: Infinity, // Keep stale cache forever for offline access
```

when online, we'd be able to refetch data every 5 mins.
when going offline, since no garbage collection would be done, we'd be able to still keep the data. 

Some more info on this, now on persistQueryClient docs: https://tanstack.com/query/latest/docs/framework/react/plugins/persistQueryClient#how-it-works. Note: 
```
IMPORTANT - for persist to work properly, you probably want to pass QueryClient a gcTime value to override the default during hydration (as shown above).

It should be set as the same value or higher than persistQueryClient's maxAge option. E.g. if maxAge is 24 hours (the default) then gcTime should be 24 hours or higher. If lower than maxAge, garbage collection will kick in and discard the stored cache earlier than expected.
```

This seems ok to me. These two options for the query cache itself, together with the maxAge of 1 day for the persisted cache (lower than gcTime recommended in the docs) on persistQueryClient, would make it possible to refetch the query while online, and avoid clearing the data in cache when offline. If you see anything wrong here let me know

Reply:
@gualter-p nice investigation work :D 

It's sounds like a great solution indeed ;)